sqlframe/base/readerwriter.py

```diff
@@ -36,6 +36,7 @@
 class _BaseDataFrameReader(t.Generic[SESSION, DF]):
     def __init__(self, spark: SESSION):
         self._session = spark
+        self.state_format_to_read = None
```

Comment:
I believe making this `self.state_format_to_read: t.Optional[str] = None` will fix the mypy issue

Reply:
Thanks for hint, updated!